### PR TITLE
Add BLOCK_HAS_PAYLOADS flag

### PR DESCRIPTION
**Pull request by Warchant**

### CODE REVIEW

**Comment by Warchant** [comment]
Breaking, because adds `status` to block serialization.

File: src/blockchain/alt_block_tree.cpp [hunk 1]
```diff
@@ -46,259 +46,91 @@ bool AltTree::bootstrap(ValidationState& state) {
...
-      if (containingIndex == nullptr) {
-        return state.Invalid(
-            "ALT-duplicate-payloads-" + P::name() + "-ancestor",
-            fmt::sprintf(
-                "Ancestor block does not exist in the chain %s that we "
-                "attempted to add to block=%s.",
-                P::name(),
-                index.toPrettyString()));
+template <typename Element, typename Container>
+auto find_duplicates(BlockIndex<AltBlock>& index, Container& pop, AltTree& tree)
```
Comment (blackyblack) [inline src/blockchain/alt_block_tree.cpp:64]: findDuplicates

Comment (blackyblack) [reply src/blockchain/alt_block_tree.cpp:64]: I think index can be const here.

File: src/blockchain/alt_block_tree.cpp [hunk 2]
```diff
@@ -46,259 +46,91 @@ bool AltTree::bootstrap(ValidationState& state) {
...
+    -> decltype(pop.end()) {
+  const auto startHeight = tree.getParams().getBootstrapBlock().height;
+  Chain<BlockIndex<AltBlock>> chain(startHeight, &index);
+  std::unordered_set<std::vector<uint8_t>> ids;
+
+  const auto& storage = tree.getStorage();
+  auto newend = std::remove_if(pop.begin(), pop.end(), [&](const Element& p) {
```
Comment (blackyblack) [inline src/blockchain/alt_block_tree.cpp:71]: Pop is being modified so the function should be name smth like findDuplicatesAndFilterPop

File: src/blockchain/alt_block_tree.cpp [hunk 3]
```diff
@@ -46,259 +46,91 @@ bool AltTree::bootstrap(ValidationState& state) {
...
-    }    // end for
-
-    it = failedPayload ? payloads.erase(it) : ++it;
-  }  // end for
+template <typename Pop>
+bool searchForDuplicates(BlockIndex<AltBlock>& index,
+                         std::vector<Pop>& pop,
+                         AltTree& tree,
+                         ValidationState& state,
+                         bool continueOnInvalid) {
```
Comment (blackyblack) [inline src/blockchain/alt_block_tree.cpp:95]: It is not just duplicates search but also Pop filtering. Are you sure you want to modify pop structure?

Comment (Warchant) [reply src/blockchain/alt_block_tree.cpp:95]: Yes. It is used in mempool.

**Pull request merged**